Tests/MeshLib/TestRasterToMesh.cpp

```diff
@@ -0,0 +1,321 @@
+    ASSERT_TRUE(mesh != nullptr);
+
+    ASSERT_EQ(2 * _n_nodes, mesh->getNumberOfBaseNodes());
+
+    std::vector<std::string> names =
+        mesh->getProperties().getPropertyVectorNames();
+    ASSERT_EQ(1, names.size());
+
+    MeshLib::PropertyVector<double>* prop =
+        mesh->getProperties().getPropertyVector<double>("test");
+    ASSERT_EQ(2 * _n_pix, prop->size());
+
+    std::pair<double, double> const& bounds =
+        MeshLib::MeshInformation::getValueBounds<double>(*mesh.get(), "test");
+    ASSERT_NEAR(0, bounds.first, std::numeric_limits<double>::epsilon());
+    ASSERT_NEAR(0.07, bounds.second, std::numeric_limits<double>::epsilon());
+
+    std::vector<MeshLib::Node*> const& nodes = mesh->getNodes();
+    for (MeshLib::Node* n : nodes)
+        ASSERT_TRUE(((*n)[2] == 0) || ((*n)[2] == _spacing));
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh.get());
+    ASSERT_EQ(2 * _n_pix, n_types[6]);
+}
+
+TEST_F(RasterToMeshTest, convertRasterToHexMeshValue)
+{
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::RasterToMesh::convert(
+        *_raster.get(), MeshLib::MeshElemType::HEXAHEDRON,
+        MeshLib::UseIntensityAs::MATERIALS, "MaterialIDs"));
+    ASSERT_TRUE(mesh != nullptr);
+
+    ASSERT_EQ(2 * _n_nodes, mesh->getNumberOfBaseNodes());
+
+    std::vector<std::string> names =
+        mesh->getProperties().getPropertyVectorNames();
+    ASSERT_EQ(1, names.size());
+
+    MeshLib::PropertyVector<int>* prop =
+        mesh->getProperties().getPropertyVector<int>("MaterialIDs");
+    ASSERT_EQ(_n_pix, prop->size());
+
+    std::pair<int, int> const& bounds =
+        MeshLib::MeshInformation::getValueBounds<int>(*mesh.get(),
+                                                      "MaterialIDs");
+    ASSERT_NEAR(0, bounds.first, std::numeric_limits<double>::epsilon());
+    ASSERT_NEAR(0, bounds.second, std::numeric_limits<double>::epsilon());
+
+    std::vector<MeshLib::Node*> const& nodes = mesh->getNodes();
+    for (MeshLib::Node* n : nodes)
+        ASSERT_TRUE(((*n)[2] == 0) || ((*n)[2] == _spacing));
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh.get());
+    ASSERT_EQ(_n_pix, n_types[4]);
+}
+
+TEST_F(RasterToMeshTest, convertRasterToQuadMeshNone)
+{
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::RasterToMesh::convert(
+        *_raster.get(), MeshLib::MeshElemType::QUAD,
+        MeshLib::UseIntensityAs::NONE, "test"));
+    ASSERT_TRUE(mesh != nullptr);
+
+    ASSERT_EQ(_n_nodes, mesh->getNumberOfBaseNodes());
+
+    std::vector<std::string> names =
+        mesh->getProperties().getPropertyVectorNames();
+    ASSERT_TRUE(names.empty());
+
+    std::vector<MeshLib::Node*> const& nodes = mesh->getNodes();
+    for (MeshLib::Node* n : nodes)
+        ASSERT_TRUE((*n)[2] == 0);
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh.get());
```

Comment:
`*mesh.get()` is equal to `*mesh`. A unique pointer is also a pointer, no need to convert it before dereferencing.